src/network/tc/cake.c

```diff
@@ -0,0 +1,171 @@
+        if (r == -ENOMEM)
+                return log_oom();
+        if (r < 0)
+                return log_syntax(unit, LOG_ERR, filename, line, r,
+                                  "More than one kind of queueing discipline, ignoring assignment: %m");
+
+        c = CAKE(qdisc);
+
+        if (isempty(rvalue)) {
+                c->bandwidth = 0;
+
+                qdisc = NULL;
+                return 0;
+        }
+
+        r = parse_size(rvalue, 1000, &k);
+        if (r < 0) {
+                log_syntax(unit, LOG_ERR, filename, line, r,
+                           "Failed to parse '%s=', ignoring assignment: %s",
+                           lvalue, rvalue);
+                return 0;
+        }
+
+        c->bandwidth = k / 8;
+        qdisc = NULL;
+
+        return 0;
+}
+
+int config_parse_tc_common_applications_kept_enhanced_overhead(
+                const char *unit,
+                const char *filename,
+                unsigned line,
+                const char *section,
+                unsigned section_line,
+                const char *lvalue,
+                int ltype,
+                const char *rvalue,
+                void *data,
+                void *userdata) {
+
+        _cleanup_(qdisc_free_or_set_invalidp) QDisc *qdisc = NULL;
+        CommonApplicationsKeptEnhanced *c;
+        Network *network = data;
+        int r;
+
+        assert(filename);
+        assert(lvalue);
+        assert(rvalue);
+        assert(data);
+
+        r = qdisc_new_static(QDISC_KIND_CAKE, network, filename, section_line, &qdisc);
+        if (r == -ENOMEM)
+                return log_oom();
+        if (r < 0)
+                return log_syntax(unit, LOG_ERR, filename, line, r,
+                                  "More than one kind of queueing discipline, ignoring assignment: %m");
+
+        c = CAKE(qdisc);
+
+        if (isempty(rvalue)) {
+                c->overhead = 0;
+                qdisc = NULL;
+                return 0;
+        }
+        if (streq(rvalue, "ethernet")) {
+                c->overhead += 38;
+                c->mpu = 84;
+        } else if (streq(lvalue, "ether-vlan")) {
+                c->overhead += 4;
+        } else {
+                r = safe_atoi(rvalue, &c->overhead);
+                if (r < 0) {
+                        log_syntax(unit, LOG_ERR, filename, line, r,
+                                   "Failed to parse 'CakeOverhead=', ignoring assignment: %s",
+                                   rvalue);
+                        return 0;
+                }
+
+                if (c->overhead < -64 || c->overhead > 256) {
+                        log_syntax(unit, LOG_ERR, filename, line, 0,
+                        "Invalid 'CakeOverhead=', ignoring assignment: %s",
+                                   rvalue);
+                        return 0;
+                }
+        }
+
+        qdisc = NULL;
+
+        return 0;
```

Comment:
hmm, in some cases mpu is set here in others not. That's strange?

Reply:
yes it's not necessary mpu  to be added in all cases.  Defaults to zero as per the man page